qmail-remote.c

```diff
@@ -27,6 +27,12 @@
 #include "timeoutconn.h"
 #include "timeoutread.h"
 #include "timeoutwrite.h"
+#include "hassmtputf8.h"
+#ifdef SMTPUTF8
+#include <idn2.h>
+#include "utf8read.h"
+#include "env.h"
+#endif
```

Comment:
I'm seeing a lot of new ifdefs and it makes me a little uneasy about how "glanceable" our code is. We need some ifdefs for EAI because it's compile-time optional, but they can be (1) more isolated and (2) fewer in number. I'll give some examples of changes I'd like to see.